### ISSUE TITLE
using a different zarr store

### ISSUE THREAD

**Issue by satra** [constantinpape/z5#185]
for one of the datasets that i am looking at i need to use a NestedDirectoryStore `zs = zarr.NestedDirectoryStore(...)` 

i was wondering how i could use this in the convert_to_h5 function. otherwise it reads zeros from the source dataset.

**Comment by constantinpape** [comment]
z5 does not supported the nested chunk layout for zarr yet, see also #184. (And as a side note there are still some things moving with this in zarr itself and I will only work on support once these issues are fixed, see https://github.com/zarr-developers/zarr-python/issues/769).

> i was wondering how i could use this in the convert_to_h5 function. otherwise it reads zeros from the source dataset.

`convert_to_h5` uses https://github.com/constantinpape/z5/blob/master/src/python/module/z5py/util.py#L76 internally.
So you could just use `copy_dataset_impl` instead and pass a zarr group derived from the NestedDirectoryStore as `f_in`.

**Comment by satra** [comment]
Thanks. I just read through the zarr issue and indeed that's a situation that needs to be dealt with. In the short term I'll use the internal call directly as you suggested. I'll leave it to you whether you want to keep this issue open or close.

**Comment by satra** [comment]
@constantinpape - tried the zarr group component, but that didn't work directly - any thoughts/ideas? it seems it's expecting attributes (`compression`) that are missing in the zarr array representation. the error happens because `ds_in`, which is computed inside the function normally returns a zarr array with a `compression` attribute. and in this case the `group[key]` returns a zarr array without a compression attribute (it has a compressor attribute). i suspect this is due to some form of standardization in z5py across file formats.
 
```python
from z5py.converter import convert_to_h5, copy_dataset_impl
import hdf5plugin
import h5py as h5

z = zarr.NestedDirectoryStore("/path/to/nested/zarr")
group = zarr.Group(store=z, read_only=True)
f = h5.File("test.h5", "a")
copy_dataset_impl(group, f, "6", "6", n_threads=6, **hdf5plugin.Blosc())
```

<details> <summary> error details </summary>

```python
In [18]: copy_dataset_impl(group, f, "6", "6", n_threads=6, **hdf5plugin.Blosc())
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-7a1a17a31695> in <module>
----> 1 copy_dataset_impl(group, f, "6", "6", n_threads=6, **hdf5plugin.Blosc())

/om2/user/satra/miniconda3/envs/zarr/lib/python3.9/site-packages/z5py/util.py in copy_dataset_impl(f_in, f_out, in_path_in_file, out_path_in_file, n_threads, chunks, block_shape, dtype, roi, fit_to_roi, **new_compression)
    112     dtype = ds_in.dtype if dtype is None else dtype
    113 
--> 114     compression = new_compression.pop("compression", ds_in.compression)
    115     compression_opts = new_compression
    116 

AttributeError: 'Array' object has no attribute 'compression'
```

</details>

for the moment i hacked the function:

```python
    if not hasattr(ds_in, "compression"):
        ds_in.compression = new_compression
```

**Comment by constantinpape** [comment]
> any thoughts/ideas? it seems it's expecting attributes (`compression`) that are missing in the zarr array representation

Yes, this makes sense; I designed everything s.t. the z5py interface is compatible with hdf5 and it seems like zarr is not fully compatible here, because zarr arrays don't have the attribute `compression`.

> for the moment i hacked the function:
> 
> ```python
>     if not hasattr(ds_in, "compression"):
>         ds_in.compression = new_compression
> ```

I think that's the best solution for now.
